docs/source/contributing.rst

```diff
@@ -96,6 +98,12 @@ Submit a request::
 Working with a local Optimism instance
 --------------------------------------
 
+Before starting any local rollup, use::
+
+    git submodule update --init
+
+to pull rollup.
```

Comment:
I would move this into a new section, at the top, e.g.

```
Setting up the repository
--------------------------------------

Clone the repository:
git clone  --recurse-submodules git@github.com:beamer-bridge/beamer.git

If you already have a repository, make sure the submodules are up to date:
git submodule update
```

Reply:
So, i see one problem with the first command here. There are people that don't use SSH keys for connecting to github (don't ask me why) and therefore it might fail due to unrecognized host/public key.
In order to be sure that this command will never fail, i suggest we use the HTTPS protocol here since that is supported by default when using git.

Also i would suggest to merge these two steps into one like so:
```
Setting up the repository
--------------------------------------

Clone the repository:
git clone https://github.com/beamer-bridge/beamer.git && 
cd beamer && 
git submodule update --init
```

What do you think about this?

Reply:
> So, i see one problem with the first command here. There are people that don't use SSH keys for connecting to github (don't ask me why) and therefore it might fail due to unrecognized host/public key.
> In order to be sure that this command will never fail, i suggest we use the HTTPS protocol here since that is supported by default when using git.

Actually, that's a very good point and we should change it to use the https transport, not the git one.
Thanks for noticing, Gabriel.

I would leave the `--recurse-submodules` part though, because then you don't need to go and manually update the submodules -- the clone will do that for you.

Reply:
Alright, lets stick with the `--recurse-submodules`. 
So we can do it like:
```
Setting up the repository
--------------------------------------

Clone the repository:
git clone  --recurse-submodules https://github.com/beamer-bridge/beamer.git && cd beamer

If you already have a repository, make sure the submodules are up to date:
git submodule update

Reply:
Right, and you don't even need the `&& cd beamer` (that only made sense with the submodule update following it immediately).

Reply:
hmm, but the next steps after setting up the repository should be done inside the repo folder.

Reply:
> hmm, but the next steps after setting up the repository should be done inside the repo folder.

Yes, but the general assumption is whoever is trying to clone & use the code is capable of deducing that fact for themselves. :-)

Reply:
Well, if we go in that direction, then we should also consider the SSH -> HTTPS edit of the first command as superfluous :P

Reply:
IMO, its just a really nice feeling when things simply work by copying & pasting the commands one by one. We want anyone to be able to contribute & understand the setup, not just techies.

Reply:
> IMO, its just a really nice feeling when things simply work by copying & pasting the commands one by one. We want anyone to be able to contribute & understand the setup, not just techies.

Yeah, I see where you're coming from. On the other hand, the fact is that we simply list the requirements and expect people to be able to install them on their own. It's a tricky balance between giving people ready-to-use commands and not giving too much so that they do not see the forest for the trees.

In any case, I don't feel strongly about it so it's fine with me.

Reply:
I don't see so much difference so i will stick with one liner command and will change it to use https instead of ssh. The person who will work on this can easily use cd command imo.